packages/connections/src/AbstractConnection.js

```diff
@@ -0,0 +1,115 @@
+import { EventEmitter } from "events";
+
+/**
+ * Connection init state
+ *
+ * @constant
+ * @type {Symbol}
+ */
+const INIT = Symbol("INIT");
+
+/**
+ * Connection started state
+ *
+ * @constant
+ * @type {Symbol}
+ */
+const OPEN = Symbol("OPEN");
+
+/**
+ * Connection done state
+ *
+ * @constant
+ * @type {Symbol}
+ */
+const CLOSED = Symbol("CLOSED");
+
+/**
+ * The `AbstractConnection` provides default properties, methods, states and
+ * event-definitions. Extend this class when creating a custom connection type
+ * to make sure it's compatible with the `ConnectionManager`.
+ *
+ * Use the following events to communicate state changes:
+ *
+ * - `OPEN` when the connection actually blocks a pipe
+ * - `DATA` every time data is received from the server
+ * - `ERROR` when an error occurred
+ * - `COMPLETE` when the connection frees its pipe
+ */
+export default class AbstractConnection extends EventEmitter {
+  /**
+   * Abstract connection
+   * @param {string} url
+   */
+  constructor(url) {
+    super();
+
+    if (this.constructor === AbstractConnection) {
+      throw new Error("Can't instantiate abstract class!");
+    }
+
+    if (!url) {
+      throw new Error("Can't instantiate without given URL!");
+    }
+
+    /**
+     * @property {string}
+     * @name AbstractConnection#url
+     */
+    Object.defineProperty(this, "url", { value: url });
+
+    /**
+     * @property {number}
+     * @name AbstractConnection#created
+     */
+    Object.defineProperty(this, "created", { value: Date.now() });
+
+    /**
+     * @property {Symbol}
+     * @protected
+     * @name AbstractConnection#state
+     */
+    let state = AbstractConnection.INIT;
+    Object.defineProperty(this, "state", {
+      get: () => state,
+      set: value => {
+        if (![INIT, OPEN, CLOSED].includes(value)) {
+          throw new Error("Cant set Connection into unknown state.");
+        }
+        state = value;
+      }
+    });
+  }
+
+  static get INIT() {
+    return INIT;
+  }
+
+  static get OPEN() {
+    return OPEN;
+  }
+
+  static get CLOSED() {
+    return CLOSED;
+  }
+
+  /* Abstract Methods */
+  /* eslint-disable no-unused-vars */
+  /**
+   * Opens the connection
+   *
+   * @description This method should only be called by Connection Manager,
+   * if you call it yourself be sure you know what you're doing.
+   * Also this method should be the point where a Connection actually starts
+   * using a "Connection Pipe" of the browser.
+   *
+   * @param {string} token – Authentication token
+   */
+  open(token) {}
```

Comment:
Let's remove the argument because it's not required here. If I follow the documentation and inherit from this class then I would expect I need to call `super.open(token)` but I don't need to do that. This documentation belongs on the override in `XHRConnection`

Reply:
Yeah, this a leftover and you don't need to pass the token.

Reply:
+1 for removal, also see my comment on `XHRConnection` below.